uber/templates/macros.html

```diff
@@ -21,9 +21,33 @@
 {%- endmacro %}
 
 
-{% macro checkbox(model, field_name, id=None, label=None, label_class=None) -%}
-    {% if label %}<label class="checkbox-label{% if label_class %} {{ label_class }}{% endif %}">{% endif %}
-    <input type="checkbox" name="{{ field_name }}" id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}" value="1" {% if model[field_name] %}checked{% endif %} />
+{% macro checkbox(model, field_name, id=None, label=None, label_class=None, readonly=False, title=None) -%}
+    {% if label %}
+      <label
+        class="checkbox-label{% if label_class %} {{ label_class }}{% endif %}"
+        {% if title %}title="{{ title }}"{% endif %}>
+    {% endif %}
+    {% if readonly %}
+      <input
+        type="hidden"
+        name="{{ field_name }}"
+        id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
+        value="{%- if model[field_name] -%}1{%- else -%}0{%- endif -%}" />
+      <input
+        type="checkbox"
+        value="1"
+        disabled
+        {% if not label and title %}title="{{ title }}"{% endif %}
+        {% if model[field_name] %}checked{% endif %} />
+    {% else %}
+      <input
+        type="checkbox"
+        name="{{ field_name }}"
+        id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
+        value="1"
+        {% if not label and title %}title="{{ title }}"{% endif %}
+        {% if model[field_name] %}checked{% endif %} />
+    {% endif %}
```

Comment:
I updated our checkbox macro to support readonly checkboxes. If `readonly=True` is passed, then a disabled checkbox will be rendered with a hidden input for the actual field value. In this manner we can create readonly checkboxes, while retaining the ability to POST the correct field value back to the server.